### ISSUE TITLE
Changing Accent color in VB.Net

### ISSUE THREAD

**Issue by furiousbuilder** [ButchersBoy/MaterialDesignInXamlToolkit#36]
Hi, I'm trying to make an option in my application to let the user choose the accent color but I don't know if it's possible...
How can I change the accent color directly from a button for exemple (In VB.Net) ? 

Thanks, 
Clément.

EDIT : I must learn to read :p It is a ChangeAppStyle which can be used easily... It's solved :)
EDIT2 : It only changes MahApps accent, not Material accent :/ It's not solved

**Comment by furiousbuilder** [comment]
UP :) Is there a way to change Material Design accent in VB.Net (for exemple with a button) ? Thanks :)

**Comment by ButchersBoy** [comment]
Hi, yes there should be a way: all themes are build using dynamic resources so they can change at runtime...but I don't have an example right now.  I'll have a look when I look at your other bug, hopefully this week, but am gonna prioritize your other bug first.

**Comment by furiousbuilder** [comment]
Okay thank you @ButchersBoy :) Your work is amazing :)

**Comment by ButchersBoy** [comment]
If you like it, just hit the :star: button :wink:

**Comment by furiousbuilder** [comment]
Done ;)

**Comment by ButchersBoy** [comment]
Legend :+1:

**Comment by ButchersBoy** [comment]
![paletterswitcher](https://cloud.githubusercontent.com/assets/658107/9396115/79956848-478a-11e5-9e64-0fa300ac455d.gif)

Nearly there...got a bit of tidying up to do before I make it available...

**Comment by ButchersBoy** [comment]
OK.  First cut of palette selection is in.  

* You will need to update MaterialDesignColors
* But MaterialDesignThemes now has a (nuget) dep on MaterialDesignColors...so just update MaterialDesignThemes
* There is no palette control: there are helper classes ```SwatchesProvider``` and ```PaletteHelper```.  There is a window in one of the test projects which shows how to build a basic palette selector.  _Maybe_ I will also build one as part of the lib in the future.

...let me know - once you've plugged this in - how your dark theme issue is.

**Comment by KlioHome** [comment]
Hi. It would be nice to add a function like this new PaletteHelper().ReplacePrimaryColor (ColorName); where ColorName = "blue" for an example. Is it possible?
P.S. Thank you for your work!

**Comment by ButchersBoy** [comment]
yeah can probably do that...

**Comment by furiousbuilder** [comment]
Thanks for your amazing work ! :) But how to change the theme now ? If I do something like 
MahApps.Metro.ThemeManager.ChangeAppTheme(Application.Current, "BaseDark")
It's like before, with the bug :/

**Comment by ButchersBoy** [comment]
And you are changing the Material theme to dark at the same time using ```new PaletteHelper().SetLightDark(isDark);```?

Now I've got the API in place, can you post a little demo project?  It will save me some time, then we can get the bug cleared up once and for all...

**Comment by furiousbuilder** [comment]
You mean something like this ? 
MahApps.Metro.ThemeManager.ChangeAppTheme(Application.Current, new PaletteHelper().SetLightDark(isDark)) ?
I'm in VB.Net, I'm not really good to do the translation between C# and VB.Net, I'll create a sample project :)

**Comment by ButchersBoy** [comment]
Yeah, post a VB project, I'll figure it out, and get the TimePicker working.

**Comment by furiousbuilder** [comment]
This is a little sample I just made : http://www.mediafire.com/download/6mfzie0arh7o4s2/MaterialTemplateProject.zip

**Comment by ButchersBoy** [comment]
OK, I'll have a look over the weekend and add suggestion by @KlioHome

**Comment by furiousbuilder** [comment]
I precise that MahApps.Metro.ThemeManager.ChangeAppTheme(Application.Current, New MaterialDesignThemes.Wpf.PaletteHelper().SetLightDark(isDark:=True)) isn't working :/

**Comment by ButchersBoy** [comment]
Yeah, think you've misunderstood that one, you should do:
```
MahApps.Metro.ThemeManager.ChangeAppTheme(Application.Current, "BaseDark")
New MaterialDesignThemes.Wpf.PaletteHelper().SetLightDark(isDark:=True)
```

**Comment by furiousbuilder** [comment]
I got few errors : http://i.imgur.com/T8lFVgT.png (Syntax error here)
http://i.imgur.com/onndNoE.png (A class can't be used as an expression)

**Comment by ButchersBoy** [comment]
OK...I can't look today, but I'll have a look at your test project over the weekend.

**Comment by furiousbuilder** [comment]
I tried another code, I don't have any error at the building but I got an error when trying to change the theme 
![screenshot 21](https://cloud.githubusercontent.com/assets/10899984/9409215/19d4abd2-4819-11e5-89d8-90d35000c624.png)
Here is the error : http://pastebin.com/fqJLMByX

**Comment by ButchersBoy** [comment]
@KlioHome you can no do ```new PaletteHelper().ReplacePrimaryColor("Blue");```

**Comment by ButchersBoy** [comment]
@furiousbuilder We are finally in business with this one :smile: 

![image](https://cloud.githubusercontent.com/assets/658107/9423973/bbfbda7a-48d2-11e5-8549-84cb557fe0cd.png)

The MD Toolkit ```PaletteHelper``` will now automatically detect MahApps styles and change them for you, so in line with that, and other recent changes, I've updated your code-behind and app.xaml, as per this [gist](https://gist.github.com/ButchersBoy/01d288110ed46a76c7b1) (remember to update from nuget first).

If you're happy with this please close off this issue so I know we're done.

**Comment by KlioHome** [comment]
@furiousbuilder, thank you so much! It's amaizing!!!

**Comment by furiousbuilder** [comment]
Is this possible to have the sample you just modified please :) To understand which lines you modified :)
Thank you

**Comment by furiousbuilder** [comment]
Using you code, I have errors :/ 
![screenshot 22](https://cloud.githubusercontent.com/assets/10899984/9425725/e742cb5a-491b-11e5-8256-caf31dcb69c5.png)
The error : http://pastebin.com/8cxHqP1j

**Comment by furiousbuilder** [comment]
The problem occurs when I add this to my app.xaml : 
<ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/FlatButton.xaml" />
<ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Controls.AnimatedTabControl.xaml" />
<ResourceDictionary Source="/Resources/Icons.xaml" />

**Comment by ButchersBoy** [comment]
Works on my PC :wink: 

Uploaded code here: https://github.com/ButchersBoy/MaterialDesignInXamlToolkit_Issue36

**Comment by ButchersBoy** [comment]
Use a full pack URL:

```<ResourceDictionary Source="pack://application:,,,/MaterialTemplateProject;component/Resources/Icons.xaml" />```

**Comment by furiousbuilder** [comment]
Oh yes, thanks for your amazing work ! :D

**Comment by furiousbuilder** [comment]
I don't know if it's a bug but now, Flyouts with theme "Accent" are whites, like with this flyout : 
Controls:Flyout x:Name="ParamFlyout" CloseButtonVisibility="Hidden" Header="Paramètres" Position="Right" Width="350" AnimateOpacity="True" Theme="Accent"
I need to set the background manually

**Comment by furiousbuilder** [comment]
I think there is a bug : with this version now, when you create a Flyout with "Accent" theme it appears White in Light mode and dark in Dark mode, so the accent isn't used : 
![screenshot 24](https://cloud.githubusercontent.com/assets/10899984/9428250/5db11b72-49a6-11e5-8f41-73350927df02.png)

**Comment by ButchersBoy** [comment]
Ok, can you raise a superstar issue for this? I have similar issue with MahApps dialogs, and I am starting to prioritise now what goes into version 1.0. MahApps integration issues may go into 1.1 so I can finalise the core Material stuff. Also, if you update the test project with a pull request showing the issues we can work on and test the solution a lot quicker when I get to it.

**Comment by ygy8643** [comment]
Hi, I am also building my VB.net application using this great work, and I also want to change accent color. But when I built the test application listed above(https://github.com/ButchersBoy/MaterialDesignInXamlToolkit_Issue36) using VS2013, I got this error 

The project currently contains references to more than one version of MaterialDesignColors, a direct reference to version 1.1.3.0 and an indirect reference (through 'MaterialDesignThemes.Wpf.PaletteHelper.ReplaceAccentColor') to version 2.3.0.823. Change the direct reference to use version 2.3.0.823 (or higher) of MaterialDesignColors.

But in VS2017 the test application works fine. How can I fix this?

Thanks for your attention.
